src/main/java/com/movie_collection/dal/dao/MovieDAO.java

```diff
@@ -0,0 +1,80 @@
+package com.movie_collection.dal.dao;
+
+import com.movie_collection.be.Movie;
+import com.movie_collection.dal.ConnectionManager;
+import com.movie_collection.dal.daoInterface.IMovieDAO;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MovieDAO implements IMovieDAO {
+    private static final ConnectionManager cm = new ConnectionManager();
+
+    public void createMovie(Movie movie) throws SQLException {
+        try (Connection con = cm.getConnection()) {
+            String sql = "INSERT INTO Movie (name, rating, path, lastview) VALUES (?, ?, ?, ?)";
+            PreparedStatement pstmt = con.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);
+            pstmt.setString(1, movie.name());
+            pstmt.setDouble(2, movie.rating());
+            pstmt.setString(3, movie.path());
+            pstmt.setDate(4, movie.lastview());
+            pstmt.executeUpdate();
+        }
+    }
+
+    public void deleteMovie(int id) throws SQLException {
+        try (Connection con = cm.getConnection()) {
+            String sql = "DELETE FROM Movie WHERE id = ?";
+            PreparedStatement pstmt = con.prepareStatement(sql);
+            pstmt.setInt(1, id);
+            pstmt.executeUpdate();
+        }
+    }
+
+    public void updateMovie(Movie movie) throws SQLException {
+        try (Connection con = cm.getConnection()) {
+            String sql = "UPDATE Movie SET name = ?, rating = ?, path = ?, lastview = ? WHERE id = ?";
+            PreparedStatement pstmt = con.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);
+            pstmt.setString(1, movie.name());
+            pstmt.setDouble(2, movie.rating());
+            pstmt.setString(3, movie.path());
+            pstmt.setDate(4, movie.lastview());
+            pstmt.setInt(5, movie.id());
+            pstmt.executeUpdate();
+        }
+    }
+
+    public Movie getMovie(int id) throws SQLException {
```

Comment:
can we try to do the optional ? or I can look at it later ?

Reply:
How do you want to implement the optional?

Reply:
public Optional<Movie> getById(int id) {
        try (Connection cnn = connector.getConnection()) {
          // implement ->
            return rs.isNotEmpty()
                    ? Optional.of(result.get(0).into(Movie.class))
                    : Optional.empty();
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }

    }
    
    how about something like this `?

Reply:
maybe to be more specific what it does getMovieById since its more open for future creation of methods like getMovieByYear etc then just generic :) 
argue if u thing otherwise :)

Reply:
Implemented here
https://github.com/TomassSimko/Private-Movie-Collection/pull/7/commits/c0f83bd0555a1b0de25b3393266f4a4dfc14c615